### PR TITLE
Change `__future__` docs description to avoid confusion with concurrency futures

### PR DESCRIPTION
**Pull request by arhadthedev**
Currently, [the library index page](https://docs.python.org/3/library/index.html) describes `__future__` as:

> \_\_future__ — Future statement definitions

However, it creates confusion with futures from the concurrency domain. So this PR changes the description to:

> ~~\_\_future__ — Early access to language features created for future versions of Python~~

*Edit: replaced with a shorter wording:*

> \_\_future__ — Early access to future language features

### CODE REVIEW

File: Doc/library/__future__.rst [hunk 1]
```diff
@@ -1,5 +1,5 @@
-:mod:`__future__` --- Future statement definitions
-==================================================
+:mod:`__future__` --- Early access to language features created for future versions of Python
+=============================================================================================
 
 .. module:: __future__
    :synopsis: Future statement definitions
```
Comment (carljm) [inline Doc/library/__future__.rst]: Not sure where all the synopsis is used, but should it also match the title text?

File: Doc/library/__future__.rst [hunk 2]
```diff
@@ -1,5 +1,5 @@
-:mod:`__future__` --- Future statement definitions
-==================================================
+:mod:`__future__` --- Early access to language features created for future versions of Python
```
Comment (carljm) [inline Doc/library/__future__.rst]: "created" seems like an unnecessary word here
```suggestion
:mod:`__future__` --- Early access to language features for future versions of Python
```

**Review by carljm — COMMENTED** [review]
Looks like an improvement to me.

File: Doc/library/__future__.rst (hunk 1 above)
Comment (arhadthedev) [reply Doc/library/__future__.rst]: Thank you, addressed.

**Review by carljm — APPROVED** [review]
LGTM!

**Comment by arhadthedev** [comment]
Could anybody merge the PR if possible please? This is the last one after gh-92761 and gh-92762.

**Comment by JelleZijlstra** [comment]
I don't find the current wording unclear, and the proposed new wording is so long that it will make the page title spill into multiple lines on many screens. If another core dev has a different opinion, I won't object to merging this though.

**Comment by arhadthedev** [comment]
@JelleZijlstra Will `Early access to future language features` be better?

```diff
diff --git a/Doc/library/__future__.rst b/Doc/library/__future__.rst
index 8bd23daee7397..00912e21d89f1 100644
--- a/Doc/library/__future__.rst
+++ b/Doc/library/__future__.rst
@@ -1,8 +1,8 @@
-:mod:`__future__` --- Future statement definitions
-==================================================
+:mod:`__future__` --- Early access to future language features
+==============================================================
 
 .. module:: __future__
-   :synopsis: Future statement definitions
+   :synopsis: Early access to future language features
 
 **Source code:** :source:`Lib/__future__.py`
```

**Comment by arhadthedev** [comment]
Applied the shorter wording described above.

**Comment by arhadthedev** [comment]
Ping.

**Comment by JelleZijlstra** [comment]
Sorry, I still don't see a reason for change here. I'm leaving this open in case another core dev thinks there's room for improvement, but I'm not planning to merge anything.
